apps/ledger-live-desktop/src/renderer/modals/CreateLocalManifest/index.tsx

```diff
@@ -0,0 +1,371 @@
+import { DEFAULT_FORM, DEFAULT_VALUES } from "./defaultValues";
+
+function createLocalManifest() {
+  return (
+    <Modal
+      width={600}
+      name="MODAL_CREATE_LOCAL_APP"
+      centered
+      render={({ data, onClose }) => <FormLocalManifest data={data} onClose={onClose} />}
+    />
+  );
+}
+
+function FormLocalManifest({
+  onClose,
+}: {
+  data: { manifest?: LiveAppManifest };
+  onClose: () => void;
+}) {
+  const { addLocalManifest } = useLocalLiveAppContext();
+  const [form, setForm] = useState<LiveAppManifest>(cloneDeep(DEFAULT_FORM));
+  const [isDapp, setIsDapp] = useState("dapp" in form);
+
+  const completeManifests = useManifests({ visibility: ["complete"] });
+  const { categories } = useCategories(completeManifests);
+  const walletAPICurrencies: WalletAPICurrency[] = useWalletAPICurrencies();
+
+  DEFAULT_VALUES.categories = categories.filter((category: string) => category !== "all");
+  DEFAULT_VALUES.currencies = walletAPICurrencies.map(currency => currency.id);
+
+  const handleSwitchEthDapp = () => {
+    setIsDapp(prevState => !prevState);
+    setForm((prevState: LiveAppManifest) => {
+      !isDapp
+        ? (prevState.dapp = cloneDeep(DEFAULT_FORM.dapp) as LiveAppManifestDapp)
+        : delete prevState.dapp;
+      return prevState;
+    });
+  };
+
+  const formIsValid = useMemo(() => LiveAppManifestSchema.safeParse(form).success, [form]);
+
+  const handleChange = useCallback((path: string, value: unknown) => {
+    setForm((prevState: LiveAppManifest) => {
+      return updateObjectAtPath({ ...prevState }, path, value);
+    });
+  }, []);
+
+  const submitHandler = (e: KeyboardEvent) => {
+    e.preventDefault();
+    if (e.detail !== 0) {
+      formIsValid && addLocalManifest(form);
+      onClose();
+    }
+  };
+
+  const contentInput = useCallback(() => {
+    return Object.keys(form.content).map(uncastedContentKey => {
+      const contentKey = uncastedContentKey as keyof LiveAppManifest["content"];
+      const value = form.content[contentKey]?.en;
+      if (value === undefined) return <></>;
+
+      const parseCheck = LiveAppManifestSchema.shape.content.shape[contentKey].safeParse({
+        en: value,
+      }).success;
+      const optional = LiveAppManifestSchema.shape.content.shape[contentKey].isOptional();
+      const path = `content.${contentKey as string}.en`;
+
+      return (
+        <FormLiveAppInput
+          key={contentKey as string}
+          type={typeof value}
+          fieldName={contentKey as string}
+          value={value}
+          optional={optional}
+          parseCheck={parseCheck}
+          path={path}
+          handleChange={handleChange}
+        />
+      );
+    });
+  }, [form.content, handleChange]);
+
+  const dappInput = useCallback(() => {
+    return (
+      <>
+        <Separator></Separator>
+        <Text ff="Inter|Bold" fontSize={4}>
+          {`ETH Dapp`}
+        </Text>
+        {Object.keys(form.dapp!).map(uncastedDappKey => {
```

Comment:
You can also pass `form.dapp` as a param to avoid the `!` here
```suggestion
  const dappInput = useCallback(() => {
    return (
      <>
        <Separator></Separator>
        <Text ff="Inter|Bold" fontSize={4}>
          {`ETH Dapp`}
        </Text>
        {Object.keys(form.dapp!).map(uncastedDappKey => {
```